tests/unit_tests/dft/include/test_common.hpp

```diff
@@ -0,0 +1,228 @@
+            count++;
+            if (count > max_print) {
+                return good;
+            }
+        }
+    }
+
+    return good;
+}
+
+template <typename vec1, typename vec2>
+bool check_equal_vector(vec1 &v, vec2 &v_ref, int n, int inc, int error_mag, std::ostream &out) {
+    constexpr size_t max_print = 20;
+    int abs_inc = std::abs(inc), count = 0;
+    bool good = true;
+
+    for (int i = 0; i < n; i++) {
+        if (!check_equal(v[i * abs_inc], v_ref[i * abs_inc], error_mag)) {
+            int i_actual = (inc > 0) ? i : n - i;
+            std::cout << "Difference in entry " << i_actual << ": DPC++ " << v[i * abs_inc]
+                      << " vs. Reference " << v_ref[i * abs_inc] << std::endl;
+            good = false;
+            count++;
+            if (count > max_print) {
+                return good;
+            }
+        }
+    }
+
+    return good;
+}
+
+// Random initialization.
+template <typename fp>
+static fp rand_scalar() {
+    return fp(std::rand()) / fp(RAND_MAX) - fp(0.5);
+}
+template <typename fp>
+static std::complex<fp> rand_complex_scalar() {
+    return std::complex<fp>(rand_scalar<fp>(), rand_scalar<fp>());
+}
+template <>
+std::complex<float> rand_scalar() {
+    return rand_complex_scalar<float>();
+}
+template <>
+std::complex<double> rand_scalar() {
+    return rand_complex_scalar<double>();
+}
+
+template <>
+int32_t rand_scalar() {
+    return std::rand() % 256 - 128;
+}
+
+template <typename fp>
+void rand_vector(fp *v, int n, int inc) {
+    int abs_inc = std::abs(inc);
+    for (int i = 0; i < n; i++) {
+        v[i * abs_inc] = rand_scalar<fp>();
+    }
+}
+
+template <typename vec>
+void rand_vector(vec &v, int n, int inc) {
+    using fp = typename vec::value_type;
+    int abs_inc = std::abs(inc);
+
+    v.resize(n * abs_inc);
+
+    for (int i = 0; i < n; i++) {
+        v[i * abs_inc] = rand_scalar<fp>();
+    }
+}
+
+// Catch asynchronous exceptions.
+auto exception_handler = [](sycl::exception_list exceptions) {
+    for (std::exception_ptr const &e : exceptions) {
+        try {
+            std::rethrow_exception(e);
+        }
+        catch (sycl::exception e) {
+            std::cout << "Caught asynchronous SYCL exception:\n" << e.what() << std::endl;
+            print_error_code(e);
+        }
+    }
+};
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+void commit_descriptor(oneapi::mkl::dft::descriptor<precision, domain> &descriptor,
+                       sycl::queue queue) {
+#ifdef CALL_RT_API
+    descriptor.commit(queue);
+#else
+    TEST_RUN_CT_SELECT_NO_ARGS(queue, descriptor.commit);
+#endif
+}
+
+template <typename T, int D>
+void copy_to_host(sycl::queue sycl_queue, sycl::buffer<T, D> &buffer_in, std::vector<T> &host_out) {
+    sycl_queue.submit([&](sycl::handler &cgh) {
+        sycl::accessor accessor = buffer_in.get_access(cgh);
+        cgh.copy(accessor, host_out.data());
+    });
+    sycl_queue.wait();
```

Comment:
```suggestion
    }).wait();
```

Reply:
Done in [56d18e6](https://github.com/oneapi-src/oneMKL/pull/261/commits/56d18e60d45dc3a0b617eef414eb0e84cbe6d7c3).